hot-reload-analysis/src/main/kotlin/org/jetbrains/compose/reload/analysis/RuntimeInfo.kt

```diff
@@ -117,21 +118,40 @@ internal fun ClassInfo(classNode: ClassNode): ClassInfo? {
 internal fun RuntimeScopeInfo(classNode: ClassNode, methodNode: MethodNode): RuntimeScopeInfo? {
     val methodId = MethodId(classNode, methodNode)
     val runtimeInstructionTree = parseRuntimeInstructionTreeLenient(methodId, methodNode)
-    return createRuntimeScopeInfo(methodId, runtimeInstructionTree)
+    val isComposeEntry = isComposeEntry(methodNode)
+    return createRuntimeScopeInfo(methodId, runtimeInstructionTree, isComposeEntry)
 }
 
 
 internal fun createRuntimeScopeInfo(
     methodId: MethodId,
     tree: RuntimeInstructionTree,
+    isComposeEntry: Boolean,
 ): RuntimeScopeInfo {
     return RuntimeScopeInfo(
         methodId = methodId,
         type = tree.type,
         group = tree.group,
         methodDependencies = tree.methodDependencies(),
         fieldDependencies = tree.fieldDependencies(),
-        children = tree.children.map { child -> createRuntimeScopeInfo(methodId, child) },
-        hash = tree.codeHash()
+        children = tree.children.map { child -> createRuntimeScopeInfo(methodId, child, false) },
+        hash = tree.codeHash(),
+        isComposeEntry = isComposeEntry
     )
 }
+
+private val COMPOSE_APP_ENTRIES = setOf(
+    Ids.WindowDesktopKt.singleWindowApplication,
+    Ids.WindowDesktopKt.singleWindowApplication_default,
+    Ids.ApplicationDesktopKt.application,
+    Ids.ApplicationDesktopKt.application_default,
+    Ids.ScreenshotTestApplicationKt.screenshotTestApplication,
+    Ids.ScreenshotTestApplicationKt.screenshotTestApplication_default,
+)
```

Comment:
Idea: I think we could hardcode those, that's fine. But maybe there is an elegant solution: 
Every time a 'Non Composable' function calls into a function, passing a Composable Lambda would be illegal, right? Just unclear to me how easy this would be?